src/Commands/core/ArchiveDumpCommands.php

```diff
@@ -0,0 +1,593 @@
+    {
+        $drupalFilesPath = $this->getDrupalFilesDir();
+        $drupalFilesArchiveComponentPath = Path::join($this->archiveDir, self::COMPONENT_FILES);
+        $this->logger()->info(
+            dt(
+                'Copying Drupal files from !from_path to !to_path...',
+                ['!from_path' => $drupalFilesPath, '!to_path' => $drupalFilesArchiveComponentPath]
+            )
+        );
+
+        $excludes = $this->getRegexpsForPaths([
+            'css',
+            'js',
+            'styles',
+            'php',
+        ]);
+
+        $this->filesystem->mirror(
+            $drupalFilesPath,
+            $drupalFilesArchiveComponentPath,
+            $this->getFileIterator($drupalFilesPath, $excludes)
+        );
+
+        return $drupalFilesArchiveComponentPath;
+    }
+
+    /**
+     * Returns the path to Drupal files directory.
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    private function getDrupalFilesDir(): string
+    {
+        if (isset($this->drupalFilesDir)) {
+            return $this->drupalFilesDir;
+        }
+
+        Drush::bootstrapManager()->doBootstrap(DrupalBootLevels::FULL);
+        $drupalFilesPath = Drupal::service('file_system')->realpath('public://');
+        if (!$drupalFilesPath) {
+            throw new Exception(dt('Path to Drupal files is empty.'));
+        }
+
+        return $this->drupalFilesDir = $drupalFilesPath;
+    }
+
+    /**
+     * Returns file iterator.
+     *
+     * Excludes paths according to the list of excludes provides.
+     * Validates for sensitive data present.
+     *
+     * @param string $path
+     *   Directory.
+     * @param array $excludes
+     *   The list of file exclude rules (regular expressions).
+     *
+     * @return \Traversable
+     */
+    private function getFileIterator(string $path, array $excludes): Traversable
+    {
+        return new RecursiveIteratorIterator(
+            new RecursiveCallbackFilterIterator(
+                new RecursiveDirectoryIterator(
+                    $path,
+                    FilesystemIterator::SKIP_DOTS
+                ),
+                function ($file) use ($excludes, $path) {
+                    $localFileName = str_replace($path, '', $file);
+                    $localFileName = str_replace('\\', '/', $localFileName);
+                    $localFileName = trim($localFileName, '\/');
+
+                    foreach ($excludes as $exclude) {
+                        if (preg_match($exclude, $localFileName)) {
+                            $this->logger()->info(dt(
+                                'Path excluded (!exclude): !path',
+                                ['!exclude' => $exclude, '!path' => $localFileName]
+                            ));
+
+                            return false;
+                        }
+                    }
+
+                    $this->validateSensitiveData($file, $localFileName);
```

Comment:
Are we calling `validateSensitiveData` on every file copied? This is, perhaps, too much work, especially if that function calls `eval`.

Reply:
No, there is no overhead here. `validateSensitiveData()` analyzes files matching`sites/[subdir]/settings.php` pattern only